File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    p_tb = {}
    
    for p in participant:
        p_tb.setdefault(p, 0)
        p_tb[p] += 1
    
    for c in completion:
        p_tb[c] -= 1
        
    for k, v in p_tb.items():
        if v == 1:
            answer = k
            
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> pMap = new HashMap<>();
        for (String p : participant) {
            pMap.put(p, pMap.getOrDefault(p, 0) + 1);
        }
        
        for (String c : completion) {
            if (pMap.containsKey(c)) {
                int cnt = pMap.get(c);
                if (cnt == 1) {
                    pMap.remove(c);
                } else {
                    pMap.put(c, cnt - 1);
                }
            }
        }
        
        String answer = pMap.keySet().iterator().next();
        return answer;
    }
}